### PR TITLE
Removed deprecated code since `0.23`

### PR DESCRIPTION
**Pull request by neet-14**
### Summary
fixes issue #11144 
Changelog: Removal

### Details and comments
@1ucian0

### CODE REVIEW

File: releasenotes/notes/matplotlib_deprecate_fix_11144-914a20fc305d5432.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+upgrade:
+  - |
+    Instead use `draw` method in `~qiskit..visualization.pulse_v2.interface.py`
+deprecations:
+  - |
+    The deprecated function in :func:`~qiskit..visualization.pulse.matplotlib.py` will be remove in the future release
+fixes:
+  - |
+    Issue #11144 `Remove deprecated code since 0.23`
```
Comment (Raghav-Bell) [inline releasenotes/notes/matplotlib_deprecate_fix_11144-914a20fc305d5432.yaml]:
```suggestion
upgrade:
  - |
     :func:`qiskit.visualization.pulse.matplotlib.py` is deprecated with version 0.23, instead   
     use :func `qiskit.visualization.pulse_drawer`.

```
Also to link an issue use fixes or closes or fix etc. It fixes #11144

**Review by ElePT — COMMENTED** [review]
Thanks for giving a try to this deprecation removal, but the current changes are not doing what we were intending with the original issue. Maybe I can help clarify the purpose of this PR. 

If you look at the implementation of the [deprecation decorators](https://github.com/Qiskit/qiskit/blob/main/qiskit/utils/deprecation.py#L37), you will notice that `@deprecate_func` on top of a class `__init__` indicates that the full class is deprecated (not only the `__init__`). The goal of this PR is to remove the deprecated classes, i.e., `EventsOutputChannels`, `WaveformDrawer` and `ScheduleDrawer`. Given that these are the only classes in `qiskit/visualization/pulse/matplotlib.py`, the easiest way to go about it is to actually remove the file. If you look at all of the classes mentioned in #11144, you will notice that actually all of the classes in `qiskit/visualization/pulse` are deprecated and should be removed. I would suggest merging the changes of this PR and #11167 into one single PR where you delete `qiskit/visualization/pulse` and any related tests. 

You will then have to make sure that any references to these classes on the documentation (probably on the visualization `__init__.py` file) are also removed, or the documentation build will fail on the automated checks. Finally, you should document what you did on the release note, that is, you should explain that these classes have been removed from the codebase, that they were deprecated on qiskit-tera 0.23, and that their functionality (pulse visualization) was replaced by the interface in``qiskit.visualization.pulse_drawer``.

File: releasenotes/notes/matplotlib_deprecate_fix_11144-914a20fc305d5432.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (ElePT) [inline releasenotes/notes/matplotlib_deprecate_fix_11144-914a20fc305d5432.yaml:1]: One more note, it is very helpful for us if the release notes have descriptive names, such as `remove-mpl-pulse` (the important part is that it says "remove" and what it is removing). This is not a hard requirement, but as I said, helpful to later locate the relevant release notes. [Edit: if you apply my suggestion above, the title should be something like `remove-deprecated-visualization-pulse`]

**Comment by ElePT** [comment]
Closing this PR in favor of https://github.com/Qiskit/qiskit/pull/11213

**Pull request closed without merging**